state/prune.go

```diff
@@ -208,16 +217,19 @@ func (p *collectionPruner) deleteInBatches(iter *mgo.Iter, logTemplate string, s
 
 			now := time.Now()
 			if now.Sub(lastUpdate) >= historyPruneProgressSeconds*time.Second {
-				logger.Infof(logTemplate, deleted)
+				logger.Logf(logLevel, logTemplate, deleted)
 				lastUpdate = now
 			}
 		}
 	}
+	if err := iter.Close(); err != nil {
```

Comment:
Now we are closing the iterator both here and in the calling function?

I guess this is fine as long as calling close twice is fine.

Reply:
Close is documented as being idempotent.